File: eslint.config.m
import pluginJs from "@eslint/js";
import pluginReact from "eslint-plugin-react";

import globals from "globals";
import tseslint from "typescript-eslint";

/** @type {import('eslint').Linter.Config[]} */
export default [
  {
    ...tseslint,
    pluginReact,
    pluginJs,
    plugins: {
      "simple-import-sort": simpleImportSort,
    },
    files: ["**/*.{js,mjs,cjs,ts,jsx,tsx}"],
    ignores: ['node_modules/*'],
    rules: {
      '@typescript-eslint/no-explicit-any': 'off',
      '@typescript-eslint/no-unused-vars': 'off',
      '@typescript-eslint/ban-types': 'off',
      'no-var': 0,
      "simple-import-sort/imports": "error",
      "simple-import-sort/exports": "error"
    }
  }
];

File: eslint.config.mjs
import pluginJs from '@eslint/js'
import simpleImportSort from 'eslint-plugin-simple-import-sort'
import globals from 'globals'
import tseslint from 'typescript-eslint'
import expo from 'eslint-config-expo'

/** @type {import('eslint').Linter.Config[]} */
export default [
  {
    files: ['**/*.{js,mjs,cjs,ts,jsx,tsx}'],
    plugins: {
      'simple-import-sort': simpleImportSort,
      'expo': expo
    },
    ignores: ['node_modules/*'],
    rules: {
      '@typescript-eslint/no-explicit-any': 'off',
      '@typescript-eslint/no-unused-vars': 'off',
      '@typescript-eslint/ban-types': 'off',
      "no-unused-vars": "warn",
      'simple-import-sort/imports': 'error',
      'simple-import-sort/exports': 'error',
      "react/jsx-uses-react": "off",
      "react/react-in-jsx-scope": "off"
    },
    settings: {
    }
  },
  { languageOptions: { globals: globals.node } },
  pluginJs.configs.recommended,
  ...tseslint.configs.recommended,

]
